packages/gradients/src/GradientFactory.ts

```diff
@@ -11,14 +11,26 @@ import type { Renderer } from '@pixi/core';
  * @ignore
  * @param color - The hexadecimal form of the color.
  */
-function cssColor(color: number) {
+function cssColor(color: number | string) {
+    if(typeof color === 'string'){
+        return color;
+    }
     let string = color.toString(16);
 
-    while (string.length < 6) {
-        string = `0${string}`;
+    if(string.length === 3){
+        string = `#${string[0]}${string[0]}${string[1]}${string[1]}${string[2]}${string[2]}`;
+    }
+    else if(string.length === 8){
```

Comment:
This should really be `string.length > 6`. Otherwise it wouldn't consider the alpha of colors like `0x0F123456`.

Reply:
ahh, I see - leading '0' gets ignored. That affects your other note as well. Also, `0x000FFF77` ends up as 5 characters. Maybe we should support alpha only by passing in a valid css color string like `"rgba(0, 255, 0, 0.75)"` and leave the treatment for numbers the way it was? Otherwise, the safest thing to do might be to add an optional `alpha` property to `ColorStep`. Do you have a preference, or other ideas @ShukantPal?

Reply:
I think allowing alpha in strings is good enough.